libs/fluid-elements/provider/src/lib/token-accessor.ts

```diff
@@ -0,0 +1,146 @@
+    ...Object.getOwnPropertyNames(this.originalTokens),
+    ...Object.getOwnPropertySymbols(this.originalTokens),
+  ]);
+
+  constructor() {
+    const overrides = this._overrides;
+    const allKeys = this._allKeys;
+
+    this.tokens = new Proxy(this.originalTokens, {
+      get(
+        target: FluidDesignTokens,
+        property: PropertyKey,
+        receiver: any,
+      ): any {
+        if (typeof property !== 'string') {
+          return undefined;
+        }
+
+        return (
+          overrides.get(property) ?? Reflect.get(target, property, receiver)
+        );
+      },
+
+      set(): boolean {
+        throw new Error(
+          'Setting design tokens directly is not allowed, use setOverride() instead.',
+        );
+      },
+
+      has(_: FluidDesignTokens, property: PropertyKey): boolean {
+        return typeof property === 'string' && allKeys.has(property);
+      },
+
+      enumerate(): PropertyKey[] {
+        return [...allKeys];
+      },
+
+      ownKeys(): PropertyKey[] {
+        return [...allKeys];
+      },
+
+      getOwnPropertyDescriptor(
+        target: FluidDesignTokens,
+        property: PropertyKey,
+      ): PropertyDescriptor | undefined {
+        // Required for Object.keys(), Object.entries() etc.
+        const defaultPropertyDescriptor = Reflect.getOwnPropertyDescriptor(
+          target,
+          property,
+        );
+        if (defaultPropertyDescriptor) {
+          return defaultPropertyDescriptor;
+        }
+
+        if (typeof property === 'string' && overrides.has(property)) {
+          return {
+            enumerable: true,
+            configurable: true,
+          };
+        }
+
+        return undefined;
+      },
+    });
+  }
+
+  /**
+   * Creates or overrides a design token with the given name.
+   *
+   * If no design token with the given name exists, it will be created.
+   * Otherwise, the value will be overwritten.
+   * @param name Name of the design token
+   * @param value New value of the design token
+   */
+  setToken(name: string, value: any): void {
+    this._overrides.set(name, value);
+    this._allKeys.add(name);
+  }
+
+  /**
+   * Resets an overriden token to its initial value.
+   *
+   * If the token was created dynamically, it will be removed.
+   * If an existing token was overwritten, its value will be reset.
+   * @param name The name of the design token to reset
+   */
+  resetToken(name: string): void {
+    this._overrides.delete(name);
+    if (!(name in this.originalTokens)) {
+      this._allKeys.delete(name);
+    }
+  }
+
+  /**
+   * Creates a CSS property name for the given token name
+   * @param tokenName The JavaScript token name in UPPER_CASE_SNAKE_CASE
+   */
+  getCssPropertyName(tokenName: string): string {
+    let cachedName = cssPropertyNameCache[tokenName];
+    if (!cachedName) {
+      cachedName = cssPropertyNameCache[
+        tokenName
+      ] = `--${tokenName.toLowerCase().replace(/\_/g, '-')}`;
```

Comment:
Note that this is technically wrong since `FLUID_SPACING_MEDIUM` gets transformed to `--fluid-spacing-medium` , but the correct name per design token definition would be `--fluid-spacing--medium` with two dashes. I guess the easy workaround would be simply using `FLUID_SPACING__MEDIUM` as the JS name or changing the definitions to use a single dash.